### ISSUE TITLE
Feedback needed

### ISSUE THREAD

**Issue by allevo** [fastify/fastify-example-twitter#12]
I need a feedback from @fastify/fastify here.
***Any*** kind of feedbacks are appreciated. Really....

**Comment by jsumners** [comment]
I have a suspicion this is awesome. I will review when I have some time to dig through it.

**Comment by StarpTech** [comment]
Good work!

- Use async / await where possible I saw many mixings with callbacks, use promisify when required. It's a great project to show the state of the art. 

- Don't provide http clients it's out of topic in my opinion. The swagger backend can describe the api in detail.

**Comment by allevo** [comment]
> Use async / await where possible I saw many mixings with callbacks, use promisify when required. It's a great project to show the state of the art.

good point (https://github.com/fastify/fastify-example-twitter/issues/13)

> Don't provide http clients it's out of topic in my opinion. The swagger backend can describe the api in detail.

They are introduced for [this](https://github.com/fastify/fastify-example-twitter#split-for-building-microservices) reason. IMHO, the main aim of this project is how to use `fastify` encapsulation in order to split your project into microservices when it is needed. This is awesome feature! In my dreads I'd like to create a guide how to lead a "simple" project like this into a serie of microservices! This's why I thought the http clients.

Anyway @StarpTech this is a incoming project, so nothing is decided! Please continue to give your contribution

**Comment by mcollina** [comment]
I'm a bit lost on the need for clients in the "monolith" scenario, while they make sense in the microservices one. I think we can have two plugins that offer the same interface: one calling the microservice, and one using the direct library. Does that make sense?

**Comment by allevo** [comment]
@mcollina yeah. I like that scenario! Fixing the "client" interface, I need to figure out how to choose the interface implementations (http or method call)
